src/talker_npc.cpp

```diff
@@ -435,17 +435,25 @@ static consumption_result try_consume( npc &p, item &it, std::string &reason )
             reason = _( "It doesn't look like a good idea to consume this…" );
             return REFUSED;
         } else {
+
+            if( to_eat.rotten() ) {
```

Comment:
As per @Qrox's suggestion, NPCs with saprovore trait should accept rotten food.
Declare `trait_SAPROVORE` as
```c++
static const trait_id trait_SAPROVORE( "SAPROVORE" );
```
at the beginning of the file, and change the condition to
```suggestion
            if( !p.as_character()->has_active_mutation( trait_SAPROVORE ) && to_eat.rotten() ) {
```
to allow them eat rotten food from the avatar.